### DISCUSSION TITLE
透视表中有多个列维度字段如何隐藏其中一个列维度字段 ？

### DISCUSSION THREAD

**Discussion by Create-Peace** [antvis/S2#2591] [Q&A]
透视表如何隐藏列维度字段，因为涉及一些动态样式，所以不能在数据中真正的过滤点这些字段，只是不渲染出来，query和id中还是需要显示这些字段

**Comment by lijinke666** [comment]
https://s2.antv.antgroup.com/manual/advanced/interaction/hide-columns#2-%E9%80%8F%E8%A7%86%E8%A1%A8

**Reply by Create-Peace** [reply]
写一个在线demo，发现不能生效
https://stackblitz.com/edit/react-nn3hwx?file=src%2Findex.js,public%2Findex.html

**Reply by lijinke666** [reply]
能不能认真的看下文档啊, 你提的很多问题, 文档和示例都能解答你 

1. 首先隐藏列头现有的能力都是针对于 叶子节点, 所以你配置一下你想隐藏的列维度对应的叶子节点的就好.
2. 其次文档说的很清楚了, 透视表是配置 id, 你配一个 ['type', 'sub_type'] 肯定不生效啊.

![image](https://github.com/antvis/S2/assets/21015895/97c9c866-1a51-465e-8903-20dcaf8af633)

**Reply by Create-Peace** [reply]
实在不好意思，我理解能力太差了，我是想隐藏类别和子类别字段，把这两层级隐藏掉
![image](https://github.com/antvis/S2/assets/12973954/d2ffe872-1678-445b-b904-f453f34940c6)
我试了但是没有成功，还请再指点一下？

**Reply by lijinke666** [reply]
哦哦, 你描述是多个列头, 隐藏部分, 可能你没描述清楚, 或者我没理解到你的意思.

如果是隐藏整个列头 (红框区域的话) 可以将高度设置为 `0`, 这样对应的角头和列头就隐藏了. [这里的文档有提现](https://s2.antv.antgroup.com/manual/advanced/custom/cell-size#%E9%9A%90%E8%97%8F%E5%88%97%E5%A4%B4), 后面在 [隐藏列头] 章节也补充下好了.

```ts
import { EXTRA_FIELD } from '@antv/s2'

const s2Options = {
  style: {
    colCell: {
      height: 0,
      // 如果 [数量] 也不希望展示可以不加下面的配置.
      heightByField: {
        [EXTRA_FIELD]: 30,
      },
    },
  }
}
```

![image](https://github.com/antvis/S2/assets/21015895/cf772a36-12d6-4f2c-b53b-91ef0393b893)

**Reply by Create-Peace** [reply]
实在不好意思，我又一次没有描述清楚，需求是隐藏列维度是不全部列维度，可能隐藏一个或者多个列维度，有可能隐藏中间的列维度， 比如只隐藏 子类别字段，
![image](https://github.com/antvis/S2/assets/12973954/47a3d259-406b-45b5-8b59-53edb9d90b58)

**Reply by lijinke666** [reply]
一样的哈, 多看看文档, 多尝试下, 你想哪几列, 就用 `heightByField: { [field]: 0 }` 就好了, https://s2.antv.antgroup.com/manual/advanced/custom/cell-size#%E8%B0%83%E6%95%B4%E5%88%97%E5%A4%B4%E5%8D%95%E5%85%83%E6%A0%BC%E5%AE%BD%E9%AB%98

```ts
const s2Options = {
  style: {
    colCell: {
      heightByField: {
        type: 0,
        sub_type: 0,
      },
    },
  }
}
```
